Move Superscript/Subscript functions into their own module

dbolack-ab opened the pull request:
## Description

Moves the `^superscript^` and `^^subscript^^` extension into its own NPM module and swaps it in.

https://www.npmjs.com/package/marked-subsuper-text

## Related Issues or Discussions

- Related to #4035 

## QA Instructions, Screenshots, Recordings

### Reviewer Checklist

*Reviewers, refer to this list when testing features, or suggest new items *
- [ ] Verify new features are functional
  - [ ] Superscripts still work
  - [ ] subscripts still work
- [ ] Check for code legibility and appropriate comments

calculuschild commented:
We should put all of these npm packages under a group organization so we can all access them.

I am a member of the "naturalcrit" organization but not the owner (belongs to Scott Tolksdorf). I can see if he will turn over ownership.

In the mean time I also created a "Homebrewery" organization. I can invite you as members so you can publish packages under it.

calculuschild commented on package.json:
```diff
@@ -114,6 +114,7 @@
     "marked-extended-tables": "^1.1.0",
     "marked-gfm-heading-id": "^4.0.1",
     "marked-smartypants-lite": "^1.0.3",
+    "marked-subsuper-text": "^1.0.1",
```
Is this package live? I can't see it in NPM.

dbolack-ab replied:
Should be? I can see it and pull it.

https://www.npmjs.com/package/marked-subsuper-text ?

calculuschild replied:
Weird. It doesn't show up when I search for it. But I can open it. Will review over there as well as part of this PR.

calculuschild commented on shared/naturalcrit/markdown.js:
```diff
@@ -333,35 +334,6 @@ const mustacheInjectBlock = {
 };
 
-const superSubScripts = {
-	name  : 'superSubScript',
```
I see the package is named "subSuper" but here it is superSub. Let's make it consistent either way.

dbolack-ab replied:
Whoops. That's an error. Should be subSuper.

dbolack-ab replied:
Where I perform the import is it named `MarkedSubSuperText`

dbolack-ab replied:
I have renamed the block type in the module for consistency.

calculuschild commented:
We should also move our test cases over to the package. Minus any test cases that are checking the interaction between multiple extensions.

dbolack-ab commented:
> We should also move our test cases over to the package. Minus any test cases that are checking the interaction between multiple extensions.

I think this particular extension squeaked through without tests. The module does have some, however.

calculuschild commented:
I added some issues to your package.

dbolack-ab commented:
> I added some issues to your package. Once those are handled I think this can merge.

I've updated for those. Let me know when you think it's good and I'll increment versions.

calculuschild commented:
@dbolack-ab Added a comment to the Readme.md issue. Needs one more tweak.

calculuschild commented:
Ok, looks good now. Do you need to re-publish the package and update the version number here?

calculuschild commented:
Ah I see you already did. Merging now!
